pio/spi/spi_flash.c

```diff
@@ -122,10 +125,12 @@ int main() {
                  31.25f,  // 1 MHz @ 125 clk_sys
                  false,   // CPHA = 0
                  false,   // CPOL = 0
-                 PIN_SCK,
-                 PIN_MOSI,
-                 PIN_MISO
+                 PICO_DEFAULT_SPI_SCK_PIN,
+                 PICO_DEFAULT_SPI_TX_PIN,
+                 PICO_DEFAULT_SPI_RX_PIN
     );
+    // Make the SPI pins available to picotool
+    bi_decl(bi_3pins_with_func(PICO_DEFAULT_SPI_RX_PIN, PICO_DEFAULT_SPI_TX_PIN, PICO_DEFAULT_SPI_SCK_PIN, GPIO_FUNC_PIO0));
```

Comment:
I used `GPIO_FUNC_PIO0` rather than `GPIO_FUNC_SPI` because PIO0 is what is _actually_ controlling the pins. But if the intention is for binary_info to be more "semantic" than "factual" (IYSWIM) then I'm happy to change this to `GPIO_FUNC_SPI`.

Reply:
ah gotcha; ok so these will want to be split out into 3pins_with_names, or 4pins_with_names... picotool telling you a pin is for PIO0 isn't much help